src/data/enrollments/EnrollmentsD2Repository.ts

```diff
@@ -0,0 +1,69 @@
+import _ from "lodash";
+
+import { D2Api } from "types/d2-api";
+
+import { Id } from "domain/entities/Base";
+import { Async } from "domain/entities/Async";
+import { Enrollment } from "domain/entities/enrollments/Enrollment";
+
+import {
+    EnrollmentsRepository,
+    EnrollmentsRepositoryParams,
+    TrackerResponse,
+} from "domain/repositories/enrollments/EnrollmentsRepository";
+
+export class EnrollmentsD2Repository implements EnrollmentsRepository {
+    constructor(private api: D2Api) {}
+
+    async getList(params: EnrollmentsRepositoryParams, enrollmentsIds: Id[]): Async<Enrollment[]> {
+        const { programId, orgUnitId } = params;
+
+        const idList = enrollmentsIds.join(";");
+
+        // NOTE: looks like @eyeseetea/d2-api/api/trackerEnrollments.d.ts: Params is not interpreted correctly and marked as required
```

Comment:
I don't understand the note, which params is marked as required?

Reply:
![image](https://github.com/user-attachments/assets/5ba631be-b5f9-41f0-8646-67a977109f69)
It make sense that its something on my end because the type definition in node_modules/@eyeseetea/d2-api/api/trackerEnrollments.d.ts looks OK, but still, i get this error.

Reply:
Oh, I see, that's indeed a bug in d2-api, and hard to track down because of the cryptic error message (the source of the problem: (TrackerEnrollmentsParamsBase["orgUnit"] & PartialParams)). That should be a pick. This has been fixed in the same PR I referenced before, so nothing to worry about.